### DISCUSSION TITLE
Intersect Raycast 3D

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#653] [Feature Requests] [answered]
it would be useful if intersect raycast would work in 3D, not just a 2D raycast into a 3D world.
i'm currently messing with VR and Ultraleap Controllers and both allow pointing into the 3D world.
With the current implementation, that's not possible out of the box. I guess the math is already built in. We just need more parameters like a XYZ Origin and Ray Vector.

**Comment by pandrr** [comment]
this is now possible on dev.cables.gl, there is a switch to switch between screencoords and xyz-xyz 3d coordinates

**Comment by pandrr** [comment, marked as answer]
https://dev.cables.gl/edit/ADNbs1

**Reply by wirmachenbunt** [reply]
coolio, thanks for the update. really useful for future interaction

**Comment by inst5** [comment]
Hi
@wirmachenbunt - I have a Ultraleap 3Di controller. Could you show an example how you did integrate it with cables?

**Reply by wirmachenbunt** [reply]
you need a websocket tool to send the tracking data to the browser. check their discord channel, there is a discussion regarding this and a download link.
